R/PredictionSurv.R

```diff
@@ -148,9 +158,16 @@ PredictionSurv = R6Class("PredictionSurv",
         surv
       }
     },
-    .distrify_survmatrix = function(x) {
-      distr6::as.Distribution(1 - x, fun = "cdf",
-        decorators = c("CoreStatistics", "ExoticStatistics"))
+    .distrify_survarray = function(x) {
+      if (inherits(x, "matrix")) {
+        # create Matdist
+        distr6::as.Distribution(1 - x, fun = "cdf",
+          decorators = c("CoreStatistics", "ExoticStatistics"))
+      } else {
+        # create Arrdist
+        distr6::Arrdist$new(cdf = 1 - x, which.curve = "mean",
+          decorators = c("CoreStatistics", "ExoticStatistics"))
+      }
```

Comment:
From previous chat, we concluded that this works and sets up the curve to the median:
```
distr6::as.Distribution(1 - x, fun = "cdf",
          decorators = c("CoreStatistics", "ExoticStatistics"))
```
Great!

The problem is, after we updated the `.surv_return()` to use the `which.curve` argument, we can now calculate the `crank` based on the `mean` curve per se (or whatever is `which.curve`). What if the user wants the mean to be used also for the metrics that use the `distr` prediction? E.g. `graf` and `rcll` score will be calculated with the 'mean' and C-index that uses the `crank` with the median!

Reply:
So we need to harmonize this. If 'mean' or whatever `which.curve` is used when using `.surv_return()` should be used also here as the survival matrix/array are what is stored and every measure converts that to the appropriate `distr6` object.

Reply:
Maybe just add a new parameter with `predict` tag for `which.curve` and default to median. Add something to `@details` so people know how it works and maybe a sentence like "in theory it is possible for you to manually change `which.curve` after construction of the predicted distribution but we advise against this as it will lead to inconsistent results"

Reply:
I think we'll probably want to add a plotting method to distr6 using `matplot` so users can plot many curves at once. And we may want to add a method (that I think we previously discussed) that allows results for multiple quantiles simultaneously (though probably just using a `for` loop)

Reply:
- Code reverted (`Arrdist` uses the median)
- Doc updated
- BART learner has now a `which.curve` parameter, set to 0.5 by default

Reply:
for the plotting, I have some code with ggplot for the gallery post, it would be a nice idea to add some more nice visualizations by default in the package

Reply:
Definitely. Ultimately distr6 needs to be upgraded to use autoplot and then it would be *much* easier to add viz in mlr3proba